homeassistant/components/deconz/climate.py

```diff
@@ -229,10 +231,14 @@ def hvac_action(self) -> str:
 
         Need to be one of HVAC_MODE_*.
         """
+        if self._device.mode == THERMOSTAT_MODE_OFF:
+            return CURRENT_HVAC_OFF
+
         if self._device.state_on:
+            if self._device.mode == THERMOSTAT_MODE_COOL:
+                return CURRENT_HVAC_COOL
```

Comment:
Normally, just because the device is in cooling mode doesn't mean it's currently cooling (all the time). If the device doesn't tell us what it's doing right now we don't allow calculating or guessing that.

Reply:
Thats why I checked that the state is on in the line above. 
State is on and it is a colling device => it is currently cooling

Reply:
So state_on means that the device is active in whatever mode is set, and not just that power is on?

Reply:
Yes that's the case. According to https://dresden-elektronik.github.io/deconz-rest-doc/endpoints/sensors/#supported-sensor-types-and-states the "dry" mode still needs to be added. 
I will also extend the test file soon, @Kane610 mentioned earlier.